fs/fs.go

```diff
@@ -29,7 +29,7 @@ func NewDatastore(path string) (ds.Datastore, error) {
 
 // KeyFilename returns the filename associated with `key`
 func (d *Datastore) KeyFilename(key ds.Key) string {
-	return filepath.Join(d.path, key.String(), ObjectKeySuffix)
```

Comment:
oh this intended. I had it your way before, but the current way works like an `index.html` file. So:

```
ds.Put("/foo", "foo")
ds.Put("/foo/bar", "bar")
```
would make

```
datastore
└── foo
    └── .object
    └── bar
        └── .object
```

so we can `mv datastore/foo datastore/baz` and change all the keys `/foo* -> /baz*`.